From micah@vmware.com Fri May 19 11:26:27 2006
Message-ID: <446E0DD0.2070907@vmware.com>
Date: Fri, 19 May 2006 11:26:24 -0700
From: Micah Dowty <micah@vmware.com>
To: linux-usb-devel@lists.sourceforge.net
Cc: Greg KH <greg@kroah.com>
Subject: USB: Allow high-bandwidth isochronous packets via usbfs

This patch increases an arbitrary limit on the size of
individual isochronous packets submitted via usbfs. The
limit is still arbitrary, but it's now large enough to
support the maximum packet size used by high-bandwidth
isochronous transfers.

Signed-off-by: Micah Dowty <micah@vmware.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/devio.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/core/devio.c
+++ gregkh-2.6/drivers/usb/core/devio.c
@@ -982,7 +982,8 @@ static int proc_do_submiturb(struct dev_
 			return -EFAULT;
 		}
 		for (totlen = u = 0; u < uurb->number_of_packets; u++) {
-			if (isopkt[u].length > 1023) {
+			/* arbitrary limit, sufficient for USB 2.0 high-bandwidth iso */
+			if (isopkt[u].length > 8192) {
 				kfree(isopkt);
 				return -EINVAL;
 			}
